server.cpp

```diff
@@ -1,18 +1,22 @@
 #include <boost/asio.hpp>
+#include <boost/bind.hpp>
```

Comment:
Instead of `boost::bind` it's better to use `std::bind`, and so the `<functional>` header.